emscripten.py

```diff
@@ -534,31 +535,49 @@ def finalize_wasm(temp_files, infile, outfile, memfile, DEBUG):
 
   args = ['--detect-features', '--minimize-wasm-changes']
 
+  # if we don't need to modify the wasm, don't tell finalize to emit a wasm file
+  modify_wasm = False
+
+  # C++ exceptions and longjmp require invoke processing,
+  # https://github.com/WebAssembly/binaryen/issues/3081
+  if shared.Settings.SUPPORT_LONGJMP or shared.Settings.DISABLE_EXCEPTION_CATCHING != 1:
+    modify_wasm = True
+  if shared.Settings.WASM2JS:
+    # wasm2js requires full legalization (and will do extra wasm binary
+    # later processing later anyhow)
+    modify_wasm = True
   write_source_map = shared.Settings.DEBUG_LEVEL >= 4
   if write_source_map:
     building.emit_wasm_source_map(base_wasm, base_wasm + '.map')
     building.save_intermediate(base_wasm + '.map', 'base_wasm.map')
     args += ['--output-source-map-url=' + shared.Settings.SOURCE_MAP_BASE + os.path.basename(shared.Settings.WASM_BINARY_FILE) + '.map']
-
+    modify_wasm = True
```

Comment:
why does source map support modify the wasm?

Reply:
`--output-source-map-url` argument modifies the wasm by writing the URL into the wasm.

Reply:
Hmm .. that could be trivial transform done in python perhaps? (you could add it to this new file: https://github.com/emscripten-core/emscripten/pull/12177)

Reply:
Maybe. But as source maps are not recommended and we'll remove support for them once DWARF is stable, I think it's not worth any work on them?

Reply:
yeah I had the same idea (I was going to suggest using objcopy) but I agree that it might not be worth it. Although objcopy might be easier than python.